apis/src/components/organisms/board.rs

```diff
@@ -194,21 +255,27 @@ pub fn Board(
                 let current_point_0 = svg_point_from_touch(viewbox_ref, &evt, 0);
                 let current_point_1 = svg_point_from_touch(viewbox_ref, &evt, 1);
                 let current_distance = get_touch_distance(current_point_0.clone(), current_point_1);
-
-                batch(move || {
-                    let scale = current_distance / initial_touch_distance();
-                    let intermediate_height = viewbox_signal().height / scale;
-                    if intermediate_height > 150.0 && intermediate_height < 2000.0 {
+                let scale = current_distance / initial_touch_distance();
+                let g_bbox = get_bbox(g_ref);
+                let mut future_viewbox = viewbox_signal.get_untracked();
+                let intermediate_height = future_viewbox.height / scale;
+                future_viewbox.width /= scale;
+                future_viewbox.height /= scale;
+                future_viewbox.x =
+                    current_point_0.x() - (current_point_0.x() - future_viewbox.x) / scale;
+                future_viewbox.y =
+                    current_point_0.y() - (current_point_0.y() - future_viewbox.y) / scale;
+                if is_svg_still_visible(g_bbox, &future_viewbox)
+                    && intermediate_height > 150.0
```

Comment:
Can we make these constants and name them? In general someone who will come in and interact with the code will have an easier time understanding the code then. I don't know why you picked 150 and 2000 :smile: